internal/service/alarms/internal/alertmanager/alertmanager_test.go

```diff
@@ -53,6 +53,9 @@ var _ = Describe("Alertmanager", func() {
 			err = c.Get(ctx, client.ObjectKey{Namespace: namespace, Name: secretName}, secret)
 			Expect(err).NotTo(HaveOccurred())
 			data := strings.Replace(string(alertManagerConfig), "{{ .url }}", utils.GetServiceURL(utils.InventoryAlarmServerName), 1)
+			data = strings.Replace(data, "{{ .caFile }}", utils.DefaultServiceCAFile, 1)
```

Comment:
I'm a little confused about this technique (please fill in my knowledge gap!). From what i understand...generally content under `/kubernetes.io/serviceaccount/serviceaccount/service-ca.crt ` is scoped to Kubernetes API server right?

And instead apps' certs are managed either the kubernetes native way e.g using [cert-manager](https://cert-manager.io/) or we create our own with `openssl req -x509...` (with subj being api) and mount it using secret and link the secret in tls under Ingres?

While works i wonder if it's the "right" approach? Curious to read more on this!

Reply:
Each service account gets a pair of cert/key (if they request one).  For every service account they are all signed by the same CA.  Usually that's a default self-signed internal CA, but the end user can override that if they want (that's not relevant here because the mechanism is the same).  When the deployment (with a service account) is launched k8s auto-injects the internal root CA into a volume at the location specified by that constant.  The pod can then use that file to validate any certificate of any service account in the system.   Since we are configuring the alert manager to invoke another service account's API endpoints (us) then here we are telling it to go and use its auto-injected CA file to verify our identity. 

If we weren't another service account running on the same system then we'd either need to have our cert signed by a public CA that the alert manager already knew about or we'd have to tell the alert manager about our private CA.

Reply:
> we'd have to tell the alert manager about our private CA. 

Ah yeah that's exactly what i was thinking. ACM's observability does allow for 
https://docs.redhat.com/en/documentation/red_hat_advanced_cluster_management_for_kubernetes/2.12/html-single/observability/index#mount-secrets-alertmanager

So we with this no need for a key file. Also does that mean MultiClusterObservability config don't need to be changed? 

```
advanced:
 alertmanager:
   secrets: ['tls']
 ```